Frameworks/Blockcerts/CertificateV3.swift

```
@@ -0,0 +1,264 @@
+        self.issuanceDate = json["issuanceDate"] as! String
+        self.expirationDate = json["expirationDate"] as? String ?? ""
+    }
+}
+
+fileprivate enum MethodsForV3 {
+    static func parse(issuerJSON: AnyObject?) -> Issuer? {
+        // TODO support issuer as string or as object
+        var issuerId: String
+        
+        if issuerJSON is String {
+            issuerId = issuerJSON as! String
+            guard let issuerIdURL = URL(string: issuerId) else {
+                print("Issuer not defined")
+                return nil
+            }
+            return IssuerV2(id: issuerIdURL)
+        } else {
+            issuerId = issuerJSON?["id"] as! String
+        }
+        
+        guard let issuerData = issuerJSON as? [String : String],
+            let issuerURLString = issuerData["url"],
+            let issuerURL = URL(string: issuerURLString),
+            let logoURI = issuerData["image"],
+            let issuerEmail = issuerData["email"],
+            let issuerName = issuerData["name"],
+            let issuerDescription = issuerData["description"],
+            let issuerIdURL = URL(string: issuerId) else {
+                return nil
+        }
+        let logo = imageData(from: logoURI)
+        
+        return IssuerV2(name: issuerName,
+                        email: issuerEmail,
+                        image: logo,
+                        id: issuerIdURL,
+                        url: issuerURL,
+                        publicKeys: [],
+                        introductionMethod: .unknown,
+                        analyticsURL: nil)
+    }
+    
+    static func parse(recipientJSON json: [String : Any]) -> Recipient? {
+        if let recipientData = json["credentialSubject"] as? [String : Any] {
+            let publicKey = recipientData["publicKey"] as? BlockchainAddress
+            let name : String = recipientData["name"] as? String ?? recipientData["id"] as! String
```

Nit: Should this be ```recipientData["id"] as! String``` or ```recipientData["id"] as? String```? Please ignore if you meant this way - ie - let it crash if numbers are passed as id. I am not sure about context though.

---

Yeah that was meant to be this way.